crates/bevy_ecs/src/bundle.rs

```diff
@@ -842,11 +838,11 @@ impl<'w> BundleInserter<'w> {
         unsafe {
             deferred_world.trigger_on_add(new_archetype, entity, add_bundle.added.iter().cloned());
             if new_archetype.has_add_observer() {
-                deferred_world.trigger_observers(ON_ADD, entity, add_bundle.added.iter().cloned());
```

Comment:
This is the change we're looking for :D

Reply:
@alice-i-cecile @NthTensor Now curious about `trigger_on_add` and `trigger_on_insert`, should these get the same treatment? I also didn't have full context around the "whys" so maybe there's a reason to just apply it to specific methods.